pubsub/azurepubsub/azurepubsub.go

```diff
@@ -0,0 +1,438 @@
+	t, err := openTopic(ctx, topicName, connString, opts)
+	if err != nil {
+		return nil
+	}
+	return pubsub.NewTopic(t)
+}
+
+// openTopic returns the driver for OpenTopic. This function exists so the test
+// harness can get the driver interface implementation if it needs to.
+func openTopic(ctx context.Context, topicName, connectionString string, opts *TopicOptions) (driver.Topic, error) {		
+	ns, err := getSbNamespace(connectionString)
+	if err != nil {
+		return nil, err
+	}
+
+	t := &topic {
+		name: topicName,		
+		ns : ns,		
+	}
+	return t, nil
+}
+
+func createTopic(ctx context.Context, topicName, connectionString string, opts[] servicebus.TopicManagementOption) (error) {		
+	ns, err := getSbNamespace(connectionString)
+	if err != nil{
+		return err
+	}	
+	tm := ns.NewTopicManager()	
+	_, err = ensureTopic(ctx, topicName, tm, opts...)
+	return err
+}
+
+func ensureTopic(ctx context.Context, topicName string, topicManager *servicebus.TopicManager, opts ...servicebus.TopicManagementOption) (*servicebus.TopicEntity, error) {
+	te, err := topicManager.Get(ctx, topicName)
+	
+	if err == nil {
+		_ = topicManager.Delete(ctx, topicName)
+	}
+
+	te, err = topicManager.Put(ctx, topicName, opts...)
+	if err != nil {		
+		return nil, err
+	}
+	return te, nil	
+}
+
+func deleteTopic(ctx context.Context, topicName, connectionString string) (error) {	
+	ns, err := getSbNamespace(connectionString)
+	if err != nil{
+		return err
+	}	
+	tm := ns.NewTopicManager()	
+	te, err := tm.Get(ctx, topicName)
+	if te != nil {
+	 return tm.Delete(ctx, topicName)	
+	}
+	return nil
+}	
+
+func getSbNamespace(connectionString string)(*servicebus.Namespace, error){
+	// Ensure the Azure ServcieBus ConnectionString is correctly formatted.
+	// See https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas
+	nsOptions := servicebus.NamespaceWithConnectionString(connectionString)
+	return servicebus.NewNamespace(nsOptions)
+}
+
+func (t *topic) getSbTopic() (*servicebus.Topic, error){
+	return t.ns.NewTopic(t.name)
+}
+
+// SendBatch implements driver.Topic.SendBatch.
+func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) (error){		
+	sbTopic, err := t.getSbTopic()
+	if err != nil {
+		return nil
+	}
+	defer func(){
+		sbTopic.Close(ctx)
+	}()
+
+	topicSender, err := sbTopic.NewSender(ctx)
+	if err != nil {
+		return err
+	}
+	defer func(){
+		topicSender.Close(ctx)
+	}()
+	
+	var sendError error
+	for i, dm := range dms {
+		sbms :=servicebus.NewMessage(dm.Body)				
+		for k, v := range dm.Metadata {
+			sbms.Set(k, v)
+		}		
+
+		sendError := topicSender.Send(ctx, sbms)
+		if sendError != nil {
+			sendError = fmt.Errorf("azure pubsub failed to send at %v: %v", i, sendError)
+			break
+		}
+	}	
+	return sendError
+}
+
+func (t *topic) IsRetryable(err error) (bool){
+	return false
+}
+
+func (t *topic) As(i interface{}) bool{
+	p, ok := i.(**servicebus.Topic)
```

Comment:
I cleaned this up; getSBTopic would return an error if we were using TopicOptions (a variadic opts into NewTopic). Since we are not, err will always be null. I will add the Managers as a return type; it doesn't hurt to have it.